infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationClientRegisterListener.java

```diff
@@ -211,7 +162,7 @@ protected CorfuStoreMetadata.Timestamp performFullSync() {
         // TODO (V2 / Chris): In next PR this listener will be only for client register/unregister, remember to avoid
         //  clearing all the in-memory fields in onClientListenerResume method in next PR.
         CorfuStoreMetadata.Timestamp timestamp = configManager.onClientListenerResume();
-        configManager.generateConfig(sessionManager.getSessions());
+        configManager.generateConfig(sessionManager.getSessions(), false);
         sessionManager.getSessions().forEach(session -> {
             SnapshotSyncUtils.enforceSnapshotSync(session, corfuStore);
```

Comment:
If a new client is identified, we need to only generate the config for it.  Registration will internally create the sessions for it, trigger a negotiation and snapshot sync.

Snapshot sync should not be triggered for all sessions.

Reply:
There are 2 cases when performFullSync will get invoked:
1. LR restart - in this case, negotiation identifies which destination clusters need a snapshot sync.  But Snapshot sync will not get triggered if there was a group-destination change only.  In that case, forced snapshot sync to all sessions with this replication model should be triggered.  As we do not persist the mapping, we cannot identify which sessions underwent a mapping change.

2. Error on the listener due to a trimmed exception - here, the existing in-memory mapping is available.  Snapshot sync should be triggered for sessions which underwent a change in the mapping.

A check for changed sessions will cover both cases.

Reply:
the restart path will not be handled by this listener, it will just go through the normal discovery service and session manager path

Reply:
Yes that's right.  I am unable to recall why I added that case here.

For the 2nd case, we can avoid full sync with sessions for which there was no change, right?  It can be done in a follow-up PR.